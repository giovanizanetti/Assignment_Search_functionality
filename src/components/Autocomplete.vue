<template>
  <div>
    <b-list-group v-for="result in results" :key="result.question_id">
      <b-link
        class="text-decoration-none"
        @click="$emit('select', result.name)"
      >
        <b-list-group-item class="d-flex px-4 bg-dark text-light">{{
          result.name
        }}</b-list-group-item></b-link
      >
    </b-list-group>
  </div>
</template>

<script>
export default {
  name: "Autocomplete",
  props: {
    results: {
      type: Array,
      required: true,
    },
  },
};
</script>
