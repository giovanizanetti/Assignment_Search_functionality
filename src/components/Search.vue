<template>
  <div class="container">
    <div class="container">
      <!-- <label for="input-with-list">Type to search</label> -->
      <b-form-input
        placeholder="Type to search"
        class="container"
        v-model="query"
        @keypress.enter="handleSearch"
        list="input-list"
        id="input-with-list"
      ></b-form-input>
      <!-- <b-form-datalist id="input-list" :options="results"></b-form-datalist> -->
    </div>
    <div class="container my-3">
      <b-list-group v-for="result in results" :key="result.question_id">
        <ListItem :result="result" />
        <!-- <a :href="result.link"
          ><b-list-group-item v-on:mouseover="active = !active">{{ result.title }}</b-list-group-item></a
        > -->

        <!-- <b-list-group-item active>Dapibus ac facilisis in</b-list-group-item>
        <b-list-group-item>Morbi leo risus</b-list-group-item>
        <b-list-group-item>Porta ac consectetur ac</b-list-group-item>
        <b-list-group-item>Vestibulum at eros</b-list-group-item> -->
      </b-list-group>
    </div>
    <b-link @click="handleLoadMore" href="#foo">Load more...</b-link>
  </div>
</template>

<script>
import axios from 'axios'
import { BASE_URL } from '../config/constants'
import ListItem from './ListItem.vue'
// import debouce from 'lodash.debounce'
// import { getSearchResults } from '../api'

export default {
  name: 'Search',
  components: {
    ListItem,
  },
  data() {
    return {
      results: null,
      query: '',
      loading: false,
      page: 1,
    }
  },

  watch: {
    // whenever question changes, this function will run
    query: function () {
      console.log(this.query)
      // this.debouncedGetAnswer()
      // this.answer = 'Waiting for you to stop typing...'
      // this.debouncedGetAnswer()
    },
    page: function () {
      console.log('whatchinfg')
      this.handleSearch()
    },
  },

  created: function () {
    this.handleSeach()
    // this.debouncedGetAnswer = debouce(getAutoCompleteResults(this.query), 500)
  },
  mounted() {},
  methods: {
    fetchSugestions() {
      axios
        .get(`${BASE_URL}inname=${this.autocompleteQuery}&tagged=${this.tag}`)
        .then((response) => (this.response = response.data.items.map((item) => item.title)))
        .catch((error) => console.error(error))
        .finally(() => console.log('FINALLY'))
    },
    handleSearch() {
      this.loading = true
      axios
        .get(`${BASE_URL}&tagged=${this.query}&${this.page}`)
        // .then((response) => (this.response = response.data.items.map((item) => item.title)))
        .then((response) => (this.results = response.data.items))
        .catch((error) => console.error(error))
        .finally(() => console.log('FINALLY'))
      this.loading = false
      // this.query = ''
    },
    handleLoadMore() {
      this.page++
      this.query = ''
      // this.handleSearch()
    },
  },
}
</script>

<style></style>
