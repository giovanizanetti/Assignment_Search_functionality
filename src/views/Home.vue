<template>
  <div class="container">
    <img alt="Vue logo" src="../assets/logo.png" />
    <Search class="p-3" />
  </div>
</template>

<script>
import Search from '@/components/Search.vue'

export default {
  name: 'Home',
  components: {
    Search,
  },
}
</script>
