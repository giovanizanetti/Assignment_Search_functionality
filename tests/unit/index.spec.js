import { shallowMount } from "@vue/test-utils";
import Home from "@/views/Home";
import Search from "@/components/Search";
import Autocomplete from "@/components/Autocomplete";

describe("HelloWorld.vue", () => {
  describe("Home page", () => {
    it("renders without creashing", () => {
      const homeView = shallowMount(Home);
      expect(homeView).toBeTruthy();
    });
  });

  describe("Search component", () => {
    it("renders without crashing", () => {
      const searchComponent = shallowMount(Search);
      expect(searchComponent).toBeTruthy();
    });
  });

  describe("Autocomplete component", () => {
    it("renders without crashing", () => {
      const results = [
        {
          has_synonyms: true,
          is_moderator_only: false,
          is_required: false,
          count: 1355753,
          name: "android",
        },
        {
          has_synonyms: true,
          is_moderator_only: false,
          is_required: false,
          count: 409720,
          name: "node.js",
        },
      ];
      const autocomplete = shallowMount(Autocomplete, {
        propsData: { results },
      });
      console.log(Autocomplete);
      expect(autocomplete).toBeTruthy();
    });
  });
});
