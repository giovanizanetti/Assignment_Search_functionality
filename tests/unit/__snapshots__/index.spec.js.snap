// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue ListItem Mounts a vue component 1`] = `
<b-link href="https://stackoverflow.com/questions/70145607/fabricjs-how-to-double-click-a-pictureor-rect-to-edit-text-and-name-it">
  <b-list-group-item class="pb-0 m-2"><strong>fabricjs how to double-click a picture(or rect) to edit text and name it</strong>
    <div class="d-flex justify-content-between align-items-center">
      <div class="d-flex align-items-center py-2">
        <b-avatar size="sm" variant="info" src="https://www.gravatar.com/avatar/bf1c05f996518af72f17a8ce45ec01b9?s=256&amp;d=identicon&amp;r=PG" class="mx-2"></b-avatar> <span class="m-2 mr-auto">standly</span>
      </div>
      <div class="px-2">28/11/2021</div>
    </div>
  </b-list-group-item>
</b-link>
`;

exports[`Home component ListItem Mounts a vue component 1`] = `
<a href="https://stackoverflow.com/questions/70145607/fabricjs-how-to-double-click-a-pictureor-rect-to-edit-text-and-name-it" target="_self" class="">
  <div class="list-group-item pb-0 m-2"><strong>fabricjs how to double-click a picture(or rect) to edit text and name it</strong>
    <div class="d-flex justify-content-between align-items-center">
      <div class="d-flex align-items-center py-2"><span class="b-avatar mx-2 b-avatar-sm badge-info rounded-circle"><span class="b-avatar-img"><img src="https://www.gravatar.com/avatar/bf1c05f996518af72f17a8ce45ec01b9?s=256&amp;d=identicon&amp;r=PG" alt="avatar"></span>
        <!----></span> <span class="m-2 mr-auto">standly</span>
      </div>
      <div class="px-2">28/11/2021</div>
    </div>
  </div>
</a>
`;
